packages/tinacms/src/toolkit/fields/plugins/mdx-field-plugin/plate/components/plate-ui/toolbar.tsx

```diff
@@ -0,0 +1,168 @@
+)
+
+export const ToolbarSeparator = withCn(
+  ToolbarPrimitive.Separator,
+  'my-1 w-px shrink-0 bg-border'
+)
+
+const toolbarButtonVariants = cva(
+  cn(
+    'inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50',
+    '[&_svg:not([data-icon])]:size-5'
+  ),
+  {
+    defaultVariants: {
+      size: 'sm',
+      variant: 'default',
+    },
+    variants: {
+      size: {
+        default: 'h-10 px-3',
+        lg: 'h-11 px-5',
+        sm: 'h-9 px-2',
+      },
+      variant: {
+        default:
+          'bg-transparent hover:bg-muted hover:text-muted-foreground aria-checked:bg-accent aria-checked:text-accent-foreground',
+        outline:
+          'border border-input bg-transparent hover:bg-accent hover:text-accent-foreground',
+      },
+    },
+  }
+)
+
+const ToolbarButton = withTooltip(
+  // eslint-disable-next-line react/display-name
+  React.forwardRef<
+    React.ElementRef<typeof ToolbarToggleItem>,
+    {
+      isDropdown?: boolean
+      pressed?: boolean
+      showArrow?: boolean
+    } & Omit<
+      React.ComponentPropsWithoutRef<typeof ToolbarToggleItem>,
+      'asChild' | 'value'
+    > &
+      VariantProps<typeof toolbarButtonVariants>
+  >(
+    (
+      {
+        children,
+        className,
+        isDropdown = true,
+        showArrow,
+        pressed,
+        size,
+        variant,
+        ...props
+      },
+      ref
+    ) => {
+      return typeof pressed === 'boolean' ? (
+        <ToolbarToggleGroup
+          disabled={props.disabled}
+          type="single"
+          value="single"
+        >
+          <ToolbarToggleItem
+            className={cn(
+              toolbarButtonVariants({
+                size,
+                variant,
+              }),
+              isDropdown && 'my-1 justify-between pr-1',
+              className
+            )}
+            ref={ref}
+            value={pressed ? 'single' : ''}
+            {...props}
+          >
+            {isDropdown && showArrow ? (
+              <>
+                <div className="flex flex-1">{children}</div>
+                <div>
+                  <Icons.arrowDown className="ml-0.5 size-4" data-icon />
+                </div>
+              </>
+            ) : (
+              children
+            )}
+          </ToolbarToggleItem>
+        </ToolbarToggleGroup>
+      ) : (
+        <ToolbarPrimitive.Button
+          className={cn(
+            toolbarButtonVariants({
+              size,
+              variant,
+            }),
+            isDropdown && 'pr-1',
+            className
+          )}
+          ref={ref}
+          {...props}
+        >
+          {children}
+        </ToolbarPrimitive.Button>
+      )
+    }
+  )
+)
+ToolbarButton.displayName = 'ToolbarButton'
+
+export { ToolbarButton }
```

Comment:
**question**: Why is `ToolbarButton` not directly exported?

Reply:
Ensures "ToolbarButton" is the displayName of this component. Taken from platejs. If we try export inline displayName will be `undefined`